### ISSUE TITLE
use global instead of window

### ISSUE THREAD

**Issue by fibo** [NaturalIntelligence/strnum#7]
Hi, 

*strnum* package is a dependency of `fast-xml-parser` which is a dependency of `@aws-sdk/client-s3`.

When I try to bundle with esbuild an AWS lambda that reads/write to S3 I get an error caused by `window.parseInt` (used in *strnum* implementation).

Could it be possible to use `globalThis.parseInt` ?

**Comment by amitguptagwl** [comment]
Though I have fixed it on my local, can you please check whether your node js version supports `parseInt` and `Number.parseInt`?

**Comment by fibo** [comment]
Yes, for example Node v16 supports both `parseInt` and `Number.parseInt`.

If possible, please remove `window` or use `globalThis` or whatever you think it is worth to support isomorphic JS.

Thanks

**Comment by kuubeu** [comment]
This polyfill:
```js
if (!Number.parseInt && window.parseInt) {
    Number.parseInt = window.parseInt;
}
if (!Number.parseFloat && window.parseFloat) {
    Number.parseFloat = window.parseFloat;
}
```
is unnecessary, since [`Number.parseInt()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/parseInt#browser_compatibility) and [`Number.parseFloat()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/parseFloat#browser_compatibility) are supported in all environments[^1] supporting [`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let#browser_compatibility), which is already in use in the library

I'll remove it in the next release

[^1]: Opera added support for `let` before the `Number` methods

**Comment by amitguptagwl** [comment]
@kuubeu I believe the below code is working perfectly fine. Let me know the issues if it is causing
```js
function parse_int(numStr, base){
    //polyfill
    if(parseInt) return parseInt(numStr, base);
    else if(Number.parseInt) return Number.parseInt(numStr, base);
    else if(window && window.parseInt) return window.parseInt(numStr, base);
    else throw new Error("parseInt, Number.parseInt, window.parseInt are not supported")
}
```

**Comment by kuubeu** [comment]
What I'm saying is that this whole polyfill is unnecessary. You can just use the global `parseInt` like normal, and if it's not supported, the runtime will throw its own error. Patching this by introducing a new function is only adding unnecessary complexity and overhead

**Comment by amitguptagwl** [comment]
We've already tested it in past when the original code belongs to fast-xml-parser. I kept the polyfil after split to maintain those fixes in place. You may find the reason in faxt-xml-parser commit history may be.

**Comment by kuubeu** [comment]
But the way the new polyfill is implemented basically always uses the global `parseInt` (unless it's overridden) because it's supported everywhere. What @fibo was suggesting was to use `globalThis.parseInt` to guarantee it's not overridden

**Comment by amitguptagwl** [comment]
I agree. I'll check that in detail. Since I've less resources to test, I worry if this should not break on any particular browser or machine with different node js version.

**Comment by kuubeu** [comment]
`globalThis` has only been supported for 5 years, so that might have to be polyfilled. Otherwise I'd suggest sticking with just using `parseInt`, without any helper functions
